### PR TITLE
chore: use upstream logdedup processor

### PR DESCRIPTION
**Pull request by colelaven**
### Proposed Change
Remove bindplane-agent specific log dedupe processor, it is now present upstream.

##### Checklist
- [x] Changes are tested
- [x] CI has passed

### CODE REVIEW

File: docs/processors.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Below is a list of supported processors with links to their documentation pages.
 | Group by Trace Processor                | [groupbytraceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.109.0/processor/groupbytraceprocessor/README.md) |
 | Kubernetes Attributes Processor         | [k8sattributesprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.109.0/processor/k8sattributesprocessor/README.md) |
 | Log Count Processor                     | [logcountprocessor](../processor/logcountprocessor/README.md) |
-| Log DeDuplication Processor             | [logdeduplicationprocessor](../processor/logdeduplicationprocessor/README.md) |
+| Log DeDuplication Processor             | [logdeduplicationprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/logdedupprocessor/README.md) |
```
Comment (colelaven) [inline docs/processors.md]: looks like all the other upstream readme links point to v0.109.0, is there a reason we wouldn't just point them to main?

File: docs/processors.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Below is a list of supported processors with links to their documentation pages.
 | Group by Trace Processor                | [groupbytraceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.109.0/processor/groupbytraceprocessor/README.md) |
 | Kubernetes Attributes Processor         | [k8sattributesprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.109.0/processor/k8sattributesprocessor/README.md) |
 | Log Count Processor                     | [logcountprocessor](../processor/logcountprocessor/README.md) |
-| Log DeDuplication Processor             | [logdeduplicationprocessor](../processor/logdeduplicationprocessor/README.md) |
+| Log DeDuplication Processor             | [logdedupprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/logdedupprocessor/README.md) |
```
[the remark this answers is not in the document]
Comment (dpaasman00) [reply docs/processors.md:16]: Just so they're pointing to the version of the component that's in the current release of the version. Part of the agent release process runs a script that updates these pretty easily.

[the remark this answers is not in the document]
Comment (colelaven) [reply docs/processors.md]: Makes sense, fixed 👍

**Review by dpaasman00 — APPROVED** [review]
Works well for me

**Pull request merged**